input_network_encoder.cpp

```diff
@@ -110,19 +110,19 @@ void InputNetworkEncoder::encode(const LocalVector<Variant> &p_input, DataBuffer
 						r_buffer.add_unit_real(pending_input.operator real_t(), info.compression_level);
 						break;
 					case DataBuffer::DATA_TYPE_VECTOR2:
-						r_buffer.add_vector2(pending_input.operator Vector2(), info.compression_level);
+						_ALLOW_DISCARD_ r_buffer.add_vector2(pending_input.operator Vector2(), info.compression_level);
```

Comment:
how _ALLOW_DISCARD_ is working?

Reply:
`Vector2` is marked as no discard, so you MUST always use the return value. `_ALLOW_DISCARD_` make sure you can just ignore the return type.